ospfd/ospf_abr.c

```diff
@@ -1153,6 +1153,15 @@ static void ospf_abr_announce_rtr(struct ospf *ospf, struct prefix_ipv4 *p,
 		if (ospf_abr_nexthops_belong_to_area(or, area))
 			continue;
 
+		/* RFC3101: Do not generate ASBR type 4 LSA if NSSA ABR */
+		if (or->u.std.external_routing == OSPF_AREA_NSSA) {
+			if (IS_DEBUG_OSPF_EVENT)
+				zlog_debug(
+					"ospf_abr_announce_rtr(): do not generate LSA Type-4 %s from NSSA",
+					inet_ntoa(p->prefix));
+			continue;
+		}
+
```

Comment:
I think we also need to take care of flushing when NSSA is configured later. Right now check is only to ensure if NSSA is preconfigured and OSPF comes up.

Reply:
@Spantik I think this is already implemented... Having the type 4 LSA in the LSDB of R1 and configuring the area as NSSA between R2-R3, I can see the Type 4 LSA being flushed

```
2020/09/20 11:13:48 OSPF: ospf_abr_task(): unapprove summaries
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): Start
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): considering area 0.0.0.0
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): approved unset on summary link id 10.0.25.0
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): approved unset on asbr-summary link id 10.0.25.5   <--- ***
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): considering area 0.0.0.3
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): approved unset on summary link id 0.0.0.0
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): approved unset on summary link id 10.0.12.0
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): approved unset on asbr-summary link id 11.11.11.11
2020/09/20 11:13:48 OSPF: ospf_abr_unapprove_summaries(): Stop

r1# sh ip os database

       OSPF Router with ID (11.11.11.11)

                ASBR-Summary Link States (Area 0.0.0.0)

Link ID         ADV Router      Age  Seq#       CkSum
10.0.25.5       10.0.25.2       3600 0x80000001 0x6a40          <--- ***

                AS External Link States

Link ID         ADV Router      Age  Seq#       CkSum  Route
1.1.1.1         11.11.11.11     1039 0x800000a5 0xe801 E2 1.1.1.1/32 [0x0]
5.5.5.5         10.0.25.2       1396 0x800000a2 0x6655 E2 5.5.5.5/32 [0x0]
```

Reply:
ah I see that is happeing as part of abr_task when NSSA command issued later.